### ISSUE TITLE
Inconsistent SSH key handling in e2e tests based on provider

### ISSUE THREAD

**Issue by johnSchnake** [kubernetes/kubernetes#68687]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
Trying to run e2e test which requires an ssh key; the ability to set the key on the pod is handled differently on different providers. In AWS you can set an env var for where the file is (whole filepath), in vsphere and local you can specify a key file name but it is always relative to `~/.ssh` (which I don't want to mount into), and GCE/GKE/kubemark it is hardcoded and cant be modified at all.

See https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L3530-L3571

**What you expected to happen**:
I expected to be able to set the key value itself into an env var (instead of such a path) but even if it _must_ be a file then I expected to be able to set its whole path regardless of the provider used.

**How to reproduce it (as minimally and precisely as possible)**:
I'm using sonobuoy and have to go through `sonobuoy gen` then modify the yaml to mount a secret. In general you could do this to any node though but I have not typically ran system tests of of other pods other than through sonobuoy.

I _want_ to be able to set:
```
      - name: LOCAL_SSH_KEY
        valueFrom:
          secretKeyRef:
            name: nodekey
            key: id_rsa
```
On a pod and have that key just be used. Currently (local provider) it is trying to use it to make a filepath though. If it has to be that way then I'd want to be able to refer to the file anywhere like in AWS. It seems I should be able to make the volume/volume mount, set the env var to point to it and be done. But that will only work in AWS and currently requires I copy my secret to the ~/.ssh directory which I dont want to have to do.

**Anything else we need to know?**:
I'm happy to put in the small PR to make the providers consistent which seems like the most simple, correct path to resolve my concerns. I'd be even more thrilled to be able to create a new env var which can be used directly rather than making the signer from a file (which seems very simple to do since there is the `sshutil.MakePrivateKeySignerFromBytes` function.

**Environment**:
- Kubernetes version (use `kubectl version`):
Looking at the code on master currently. Running against
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.6", GitCommit:"4bc5e7f9a6c25dc4c03d4d656f2cefd21540e28c", GitTreeState:"clean", BuildDate:"2017-09-15T08:51:21Z", GoVersion:"go1.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"", BuildDate:"2018-06-27T08:25:31Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by johnSchnake** [comment]
@kubernetes/sig-testing

**Comment by johnSchnake** [comment]
I also have another related ticket (https://github.com/kubernetes/kubernetes/issues/68747) I'm happy to address these myself if someone from the sig concurs with my assessment of the issues/solutions.

**Comment by ixdy** [comment]
cc @kubernetes/sig-testing

**Comment by johnSchnake** [comment]
@ixdy Correct; things seem to have gotten crossed. I opened 2 issues related to SSH around the same time and put up PRs for them and because of some mentions I guess the bot got confused?

This issue (68687) was fixed by pull #68753 which has merged.
I have another issue #68747 which is fixed by pull #68792 ; I have LGTM but havent gotten approved and a go dep check.

**Comment by ixdy** [comment]
yeah, it looks like #68753 said it would fix #68747 - probably just a typo.

I left a few comments on #68792 and will approve once those are addressed.

/close
since I think this issue is now fixed.
